config/log_config.go

```diff
@@ -0,0 +1,78 @@
+package config
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"go.viam.com/rdk/logging"
+)
+
+// LoggerPatternConfig is an instance of a level specification for a given logger.
+type LoggerPatternConfig struct {
+	Pattern string `json:"pattern"`
+	Level   string `json:"level"`
+}
+
+var (
+	validLoggerNameCharPattern = `[a-zA-Z0-9_-]`
+	loggerPatternRegexp        = regexp.MustCompile(`^` + validLoggerNameCharPattern + `+$|^\*$`)
+)
+
+func validatePattern(pattern string) bool {
+	for _, sep := range strings.Split(pattern, ".") {
+		if !loggerPatternRegexp.MatchString(sep) {
+			return false
+		}
+	}
+	return true
+}
+
+// UpdateLoggerRegistry updates the logger registry if necessary  with the specified logConfig.
+func UpdateLoggerRegistry(logConfig []LoggerPatternConfig, loggerRegistry map[string]logging.Logger) (map[string]logging.Logger, error) {
+	newLogRegistry := make(map[string]logging.Logger)
+
+	for _, lpc := range logConfig {
+		if !validatePattern(lpc.Pattern) {
+			return nil, errors.New("Failed to Validate a Pattern")
+		}
+
+		var matcher strings.Builder
+		for idx, ch := range lpc.Pattern {
+			switch ch {
+			case '*':
+				if idx == len(lpc.Pattern)-1 {
+					matcher.WriteString(validLoggerNameCharPattern + `+(\.` + validLoggerNameCharPattern + `+)*`)
+				} else {
+					matcher.WriteString(validLoggerNameCharPattern + `+`)
+				}
```

Comment:
hm, one more suggestion to simplify things... maybe - let's make the following regex patterns:

```
const (
	validLoggerSectionName = `[a-zA-Z0-9_-]+`
	validLoggerSections    = validLoggerSectionName + `(\.` + validLoggerSectionName + `)*`
	validLoggerName        = `^` + validLoggerSections + `$
)

then we could change this as follows:

```suggestion
				if idx == len(lpc.Pattern)-1 {
					matcher.WriteString(validLoggerSections)
				} else {
					matcher.WriteString(validLoggerSection)
				}
```

and we use the 3rd constant for validation of the full name